src/people.rs

```diff
@@ -75,6 +91,40 @@ macro_rules! define_person_property {
     };
 }
 
+/// Defines a derived person property with the following parameters:
+/// * `$person_property`: A name for the identifier type of the property
+/// * `$value`: The type of the property's value
+/// * `[$($dependency),+]`: A list of person properties the derived property depends on
+/// * $calculate: A closure that takes the values of each dependency and returns the derived value
+#[macro_export]
+macro_rules! define_derived_person_property {
+    ($derived_property:ident, $value:ty, [$($dependency:ident),+], |$($param:ident),+| $derive_fn:expr) => {
+        #[derive(Copy, Clone)]
+        pub struct $derived_property;
+
+        impl $crate::people::PersonProperty for $derived_property {
+            type Value = $value;
+
+            fn calculate(context: &$crate::context::Context, person_id: $crate::people::PersonId) -> Self::Value {
```

Comment:
Given that initialize and calculate are the same, why not just have initialize?

Reply:
Overloading the meaning of a single function seems not ideal to me, this is more explicit